cmd/test_stores/fake_refinery.go

```diff
@@ -132,12 +132,14 @@ func (fri *FakeRefineryInstance) runDecider(opts CmdLineOptions, nodeIndex int,
 					go func(status *centralstore.CentralTraceStatus, span3 trace.Span) {
 						defer wg.Done()
 
-						trace, err := fri.Store.GetTrace(getCtx, status.TraceID)
+						traces, err := fri.Store.GetTraces(getCtx, status.TraceID)
 						if err != nil {
 							otelutil.AddException(span3, err)
 						}
-						otelutil.AddSpanField(span3, "span_count", len(trace.Spans))
-						traces = append(traces, trace)
+						for _, trace := range traces {
+							otelutil.AddSpanField(span3, "span_count", len(trace.Spans))
```

Comment:
You're trying to add `span_count` once for each trace; they're going to overwrite each other. Are you trying to record the total number of spans in the trace?